src/fixSuggestions/fixSuggestionsService.ts

```diff
@@ -0,0 +1,45 @@
+import { ShowFixSuggestionParams } from "../lsp/protocol";
+import * as vscode from "vscode";
+import { logToSonarLintOutput } from "../util/logging";
+import { SonarLintExtendedLanguageClient } from "../lsp/client";
+
+export class FixSuggestionService {
+	private static _instance : FixSuggestionService;
+
+	static init(client: SonarLintExtendedLanguageClient) {
+		FixSuggestionService._instance = new FixSuggestionService(client);
+	}
+	constructor(private readonly client: SonarLintExtendedLanguageClient) {}
+
+	static get instance() {
+		return FixSuggestionService._instance;
+	}
+
+	showFixSuggestion = async (params : ShowFixSuggestionParams) => {
+		try {
+			const fileUri = vscode.Uri.parse(params.fileUri);
+			await vscode.window.showTextDocument(fileUri);
+			const wsedit = new vscode.WorkspaceEdit();
+			params.textEdits.forEach(async (edit) => {
+				const range = new vscode.Range(edit.beforeLineRange.startLine, 0, edit.beforeLineRange.endLine, 10000);
```

Comment:
Where does this `10000` come from? (Sh|c)ould it be extracted to a constant with a descriptive name?

Reply:
It is just a placeholder so that we don't have to calculate the actual length of the line 🙃
I will put it in a constant